source/StockMarket.py

```diff
@@ -80,24 +78,16 @@ def merger(self, c1, c2):
         according to the required conditions
         '''
         if c1 in self.companies and c2 in self.companies:
-            comp_1 = Company(name=c1,
-                             stocks_num=c1.stocks_num,
-                             stock_price=c1.stock_price,
-                             comp_type=c1.comp_type)
-            comp_2 = Company(name=c2,
-                             stocks_num=c2.stocks_num,
-                             stock_price=c2.stock_price,
-                             comp_type=c2.comp_type)
-            new_comp = comp_1 + comp_2
-            if comp_1 == comp_2:
-                new_comp.name = comp_1.name
-            if comp_1 > comp_2:
-                new_comp.name = comp_1.name
+            new_comp = c1 + c2
+            if c1 == c2:
+                new_comp.name = c1.name
+            if c1 > c2:
+                new_comp.name = c1.name
             else:
-                new_comp.name = comp_2.name
+                new_comp.name = c2.name
```

Comment:
This can still be improved:

1. You can turn the first and the second if into one, can you see that the code inside is identical? think about it.
2. The optimal solution would be not changing the name of the company after its created, I want you to create the company with the right name on the first time!

Reply:
1. fixed
2. I do not think I did what you meant, but I did some change.